### PR TITLE
Feature/kids 1538 scan nfc entry page

### PR DESCRIPTION
**Pull request by Daniela510**
## Description

- Attempt to not go into the 'outside of app'  coin flow via deeplink when scanning.
- Clear state when bottomsheet is dismissed: clicking cancel in iOS, or clicking outside the bottomsheet on either platform

### CODE REVIEW

File: lib/core/app/app_router.dart [hunk 1]
```diff
@@ -87,9 +87,17 @@ class AppRouter {
...
-                state.extra != null ? state.extra as Flows : Flows.main;
-            return ChooseAmountSliderScreen(flow: flow);
+            Flows flow;
+            final bool isCoinFlow =
+                state.uri.queryParameters['isCoinFlow']?.contains('true') ??
+                    false;
+            if (isCoinFlow) {
+              flow = Flows.coin;
+              return ChooseAmountSliderScreen(flow: flow);
+            } else {
```
Comment (Daniela510) [inline lib/core/app/app_router.dart:97]: I wanted to pass the flow as an extra to keep consistency, but adding extra data when redirecting is still an open issue on the [go router package.](https://github.com/flutter/flutter/issues/105294)

File: lib/core/app/app_router.dart [hunk 2]
```diff
@@ -25,7 +25,7 @@ import 'package:go_router/go_router.dart';
 
 class AppRouter {
   static final _rootNavigatorKey = GlobalKey<NavigatorState>();
-  static bool ignoreCoinFlow = false;
+  static bool inAppCoinFlow = false;
```
Comment (pavelpavelko) [inline lib/core/app/app_router.dart:28]: can we remove this flag and use FlowsCubit state instead? 
we can add getter to the FlowsState like: 
`bool get isInAppCoin {
    return flowType == FlowType.inAppCoin;
  }
`

**Review by pavelpavelko — COMMENTED** [review]
the only comment about inAppCoinFlow flag

File: lib/core/app/app_router.dart (hunk 2 above)
Comment (Daniela510) [reply lib/core/app/app_router.dart]: Yes fair, done

**Review by pavelpavelko — APPROVED** [review]
looks good)

**Pull request merged**